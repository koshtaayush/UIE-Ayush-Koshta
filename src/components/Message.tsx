import React from 'react';
import styled from 'styled-components';

import dateFormatter from './../services/dateFormatter';
import { SENDER_NAME } from './../config/config';
import { IMessage } from './../typings/sharedInterface';

interface Props{
    message: IMessage
}

const Message: React.FC<Props> = (props) => {

    // Message prop
    const { message }  = props;

    return (
        <IncomingMessage isMyMessage={message.author === SENDER_NAME}>
            <MessageContainer isMyMessage={message.author === SENDER_NAME}>
                {!(message.author === SENDER_NAME) && <Author test-id="author">{message.author}</Author>}
                <MessageText test-id="message">{message.message}</MessageText>

                <Time isMyMessage={message.author === SENDER_NAME} test-id="timeStamp">{dateFormatter(message.timestamp)}</Time>
            </MessageContainer>
        </IncomingMessage>
    )

}

export default Message

interface IIncomingMessage{
    isMyMessage: boolean
}

interface ITime{
    isMyMessage: boolean
}

interface IMessageContainer{
    isMyMessage: boolean
}

const IncomingMessage  = styled.div<IIncomingMessage>`
    display: flex;
    align-items: center;
    justify-content: ${(props) => props.isMyMessage ? 'flex-end' : 'flex-start'};
`

const MessageContainer = styled.div<IMessageContainer>`
    border: 0.0625rem solid #6D6D7A;
    padding: 1rem;
    margin-bottom: 0.5rem;
    position: relative;
    display: inline-block;
    margin-bottom: 0.3125rem;
    font-size: 0.7em;
    padding: 0.625rem 0.625rem 0.625rem 0.75rem;
    border-radius: 0.25rem;
    max-width: 50%;
    background-color: ${(props) => props.isMyMessage ? '#FCF6C5' : '#FFFFFF'}; ;
`

const Author = styled.div`
    color: #6D6D7A;
    margin-bottom: 0.375rem;
`

const MessageText = styled.div`
    margin-bottom: 0.375rem;
    line-height: 1rem;
`

const Time = styled.div<ITime>`
    color: #6D6D7A;
    text-align: ${(props) => props.isMyMessage ? 'end' : 'start'};
`