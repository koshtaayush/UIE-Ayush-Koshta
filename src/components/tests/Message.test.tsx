import React from 'react'
import { shallow } from 'enzyme'

import Message from './../Message';

describe('Message', () => {

    it('Should match the snapshot', () => {

        let mockMessage = {
            author: "Tom",
            message: "Hello world",
            timestamp: 1629403445280,
            token: "dummyToken123",
            _id: "611eb935c1f69f001a33088f"
        }

        const component = shallow(<Message message={mockMessage} />);
        expect(component).toMatchSnapshot();
    })

    it('Should always render message and timestamp', () => {

        let mockMessage = {
            author: "Tom",
            message: "Hello world",
            timestamp: 1629403445280,
            token: "dummyToken123",
            _id: "611eb935c1f69f001a33088f"
        }

        const component = shallow(<Message message={mockMessage} />);  
        expect(component.find({ 'test-id': 'message' })).toHaveLength(1);
        expect(component.find({ 'test-id': 'timeStamp' })).toHaveLength(1);
    })
    
    it('Should render the author name in case its sent by other person', () => {

        let mockMessage = {
            author: "Tom",
            message: "Hello world",
            timestamp: 1629403445280,
            token: "dummyToken123",
            _id: "611eb935c1f69f001a33088f"
        }

        const component = shallow(<Message message={mockMessage} />);  
        expect(component.find({ 'test-id': 'author' })).toHaveLength(1);
    })
    
    it('Should not render the author name in case its sent by me', () => {

        let mockMessage = {
            author: "Ayush",
            message: "Hello world",
            timestamp: 1629403445280,
            token: "dummyToken123",
            _id: "611eb935c1f69f001a33088f"
        }

        const component = shallow(<Message message={mockMessage} />);  
        expect(component.find({ 'test-id': 'author' })).toHaveLength(0);
    })

})
