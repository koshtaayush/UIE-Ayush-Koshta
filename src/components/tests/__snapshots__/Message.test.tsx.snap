// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Should match the snapshot 1`] = `
<styled.div
  isMyMessage={false}
>
  <styled.div
    isMyMessage={false}
  >
    <styled.div
      test-id="author"
    >
      Tom
    </styled.div>
    <styled.div
      test-id="message"
    >
      Hello world
    </styled.div>
    <styled.div
      isMyMessage={false}
      test-id="timeStamp"
    >
      20 Aug 2021 1:34
    </styled.div>
  </styled.div>
</styled.div>
`;
